extensions/amp-video-iframe/1.0/amp-video-iframe.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {BUBBLE_MESSAGE_EVENTS} from '../0.1/amp-video-iframe';
+import {MIN_VISIBILITY_RATIO_FOR_AUTOPLAY} from '../../../src/video-interface';
+import {VideoBaseElement} from '../../amp-video/1.0/base-element';
+import {VideoIframe} from '../../amp-video/1.0/video-iframe';
+import {createCustomEvent} from '../../../src/event-helper';
+import {dict} from '../../../src/utils/object';
+
+/** @const {string} */
+const TAG = 'amp-video-iframe';
+
+/**
+ * @param {!Element} element
+ * @return {!Promise<IntersectionObserverEntry>}
+ */
+function getIntersectionObserverEntryAsync(element) {
+  return new Promise((resolve) => {
+    const observer = new IntersectionObserver((entries) => {
+      resolve(entries[entries.length - 1]);
+      observer.disconnect();
+    });
+    observer.observe(element);
+  });
+}
```

Comment:
I'm not sure what's the best place for this utility. It's needed for `Docking`, too.

Reply:
/cc @samouri. For now it can stay here and we can figure it out later. Overall it can go even into `dom.js` eventually.